Added test_fs_types.cpp and test_fs_name.cpp Updated makefile.am acco…

singhpratik5 opened the pull request:
Added test_fs_types.cpp for tsk/fs/fs_types.c and test_fs_name.cpp for tsk/fs/fs_name.cpp.
Updated makefile.am accordingly.

singhpratik5 committed:
Added test_fs_types.cpp and test_fs_name.cpp Updated makefile.am accordingly

singhpratik5 committed:
Merge branch 'develop' into fs-tests

singhpratik5 committed:
Update test_fs_name.cpp

Commented out cases with invalid tag.

singhpratik5 committed:
Merge remote-tracking branch 'upstream/develop' into fs-tests

singhpratik5 committed:
Removed test_fls.cpp
	modified:   Makefile.am
	deleted:    test/tsk/fs/test_fls.cpp

singhpratik5 commented:
@simsong can you take a look at this?
Should I open a PR to sleuthkit/develop?

simsong commented:
It's hard to see what's going on here because you have combined updates to develop from sleuthkit/develop with updates for your specific PR. I will create my own PR to bring you up to Develop, and then this one will be more clear.

simsong commented:
Please look at https://github.com/singhpratik5/sleuthkit/pull/3
1. Accept that PR
2. Bring this PR up to Develop.
3. Mark this PR ready for Review.

Simson

singhpratik5 committed:
Merge branch 'develop' into fs-tests

simsong commented on test/tsk/fs/test_fs_types.cpp:
```diff
@@ -0,0 +1,119 @@
+// Test for tsk_fs_type_toid function
+TEST_CASE("tsk_fs_type_toid converts TSK_TCHAR strings to filesystem type IDs", "[fs_types]") {
+    SECTION("recognizes common filesystem types") {
+        TSK_TCHAR ntfs_str[] = {'n', 't', 'f', 's', '\0'};
```
I think that you can simplify this by using the `_TSK_T()` macro.

simsong commented on test/tsk/fs/test_fs_types.cpp:
```diff
@@ -0,0 +1,119 @@
+    }
+    
+    SECTION("truncates strings longer than 15 characters") {
+        TSK_TCHAR long_str[] = {'n', 't', 'f', 's', '_', 'v', 'e', 'r', 'y', 'l', 'o', 'n', 'g', 'n', 'a', 'm', 'e', '\0'};
```
I don't think that it should be called `long_str` because it's an 8-char string under posix but a 16-char string under WIndows. 

I suggest that you name it according to what it holds, rather than its type.

simsong reviewed (changes requested):
Good going. Just a few minor tweaks.

singhpratik5 committed:
Update test_fs_types.cpp

used _TSK_T macro instead of strings as char.
Updated a str name as per it's actual behaviour

singhpratik5 reviewed (commented):
Fixed.

simsong commented:
Great. Now submit a PR from your develop to sleuthkit develop!Thanks!----Sent from my phone.On Jun 30, 2025, at 12:34 PM, singhpratik5 ***@***.***> wrote:﻿
@singhpratik5 commented on this pull request.

Fixed.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>
